pkg/notifications/stream.go

```diff
@@ -24,13 +25,19 @@ type streamImpl struct {
 
 // Consume returns the channel to retrieve notifications.
 func (s *streamImpl) Consume() <-chan *storage.Notification {
+	if s == nil {
+		return nil
+	}
```

Comment:
I suppose these changes (along with the ones for `Produce`) should be part of another PR to explicitly handle the nil case?